### PR TITLE
Update PR tests for memory_statistics

### PR DESCRIPTION
**Pull request by archigup**
Updates the test for memory_statistics to cover all its features.
Fixes a bug where the working dir was not set for comparing the output.
Removes FreeRTOSConfig.h include.
Has memory_statistics clean up its generated object files to assist in running it locally.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -99,19 +99,10 @@ jobs:
...
         with:
-          path: coreMQTT
-          config: ../.github/coreMQTT_memory_statistics_config.json
-      - name: Upload table
-        uses: actions/upload-artifact@v2
-        with:
-          name: size_table
-          path: coreMQTT/size_table.html
-          if-no-files-found: error
+          path: memory_statistics/test
```
Comment (aggarw13) [inline .github/workflows/test.yml:105]:
```suggestion
          path: ./memory_statistics/test
```

File: memory_statistics/test/test.c [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#include "test_header.h"
+#ifdef TEST_FLAG
```
Comment (aggarw13) [inline memory_statistics/test/test.c:2]: Please add a comment about the significance of this macro for the testing of the action.

File: memory_statistics/test/test.c [hunk 3]
```diff
@@ -1,4 +1,14 @@
+#include "stdint.h"
+
+// Test include directories added from memory_statistics config.
```
Comment (aggarw13) [inline memory_statistics/test/test.c:3]: (Nit)
```suggestion
// Test include directories added from test config of memory_statistics action.
```

File: memory_statistics/test/include/test_header.h [hunk 4]
```diff
@@ -0,0 +1,5 @@
+/**
+ * test_header.h
+ * This is an empty header file to test the include directories configuration of
+ * the memory_statistics action.
```
Comment (aggarw13) [inline memory_statistics/test/include/test_header.h:4]: (Nit)
```suggestion
 * This is an empty header file to test that the memory_statistics action can parse 
 * include directories from config file.
```

File: memory_statistics/memory_statistics.py [hunk 5]
```diff
@@ -28,6 +28,10 @@ def make(sources, includes, flags, opt):
     (results, _) = proc.communicate()
     print(results)
 
+    args = ['make', '-f', os.path.join(__THIS_FILE_PATH__, 'makefile'), 'clean']
```
Comment (aggarw13) [inline memory_statistics/memory_statistics.py:31]: What is the impact of these changes?

Comment (archigup) [reply memory_statistics/memory_statistics.py:31]: It removes the FreeRTOSConfig.h that was used with every build. It wasn't actually used, so this removes it.

File: memory_statistics/test/test.c (hunk 3 above)
Comment (archigup) [reply memory_statistics/test/test.c:3]: That was referring to the config provided to the action, not the specific config used for testing, Have clarified the comments.

**Review by aggarw13 — DISMISSED** [review]

File: memory_statistics/test/test.c [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#include "stdint.h"
+
+// Test include directories added from supplied memory_statistics config.
```
Comment (aggarg) [inline memory_statistics/test/test.c:3]: Please use multiline comments i.e. `/* */`. Does uncrustify not do that?

**Review by aggarg — APPROVED** [review]

**Review by yourslab — APPROVED** [review]

**Pull request merged**